app/channels/editor_channel.rb

```diff
@@ -0,0 +1,62 @@
+# The Editor Channel is opened when a user navigates to the edit page of an
+# Evidence, Issue, Note, or Card. The subscription is closed when the user
+# navigates away from the edit page. An open subscription receives updates
+# from the editor and saves them as new revisions of the resource. The channel
+# then broadcasts the resource's updated_at field back to the subscriber so
+# the form can update the original_updated_at field used for conflict
+# resolution. Although multiple users may be subscribed the channel broadcasts
+# messages out with a user scope. This means users will not receive live updates
+# of changes from other users.
+class EditorChannel < ApplicationCable::Channel
+  include ProjectScopedChannels
+  include PaperTrailActivity
+
+  attr_accessor :resource
+
+  AUTOSAVE_EVENT = 'auto-save'.freeze
+
+  def subscribed
+    reject and return unless find_resource(params)
+
+    stream_for [current_user, current_project, resource]
+  end
+
+  def save(params)
+    resource.paper_trail_event = AUTOSAVE_EVENT
+
+    if resource.update_attributes parsed_params(params['data'])
+      self.class.broadcast_to([current_user, current_project, resource], resource.reload.updated_at.to_i)
+    end
+  end
+
+  private
+
+  def find_resource(params)
+    return unless %w[evidence issue note card].include? params['resource_type']
+
+    if params['resource_type'] == 'card'
+      authorized_list_ids = List.where(board_id: current_project.boards.select(:id)).select(:id)
+      @resource ||= Card.find_by(id: params['resource_id'], list_id: authorized_list_ids)
+    else
+      @resource ||= current_project.send(params['resource_type'].pluralize).find_by id: params['resource_id']
```

Comment:
1. Why the use of `||=` here and in :39 ?

2. I'm not sure the use of `#send` is buying us much here. Yes, it's reducing the number of calls, but I'm not sure if we should go for readability of the code vs lines of code as a metric?

Should L35 and this be replaced with a `switch` statement? It's probably a smell anyway, but with a clear `switch` you know what you're reading right away, as opposed to figuring out the trickery.

We're also going to be adding at least ContentBlocks to the list in Pro too.

Reply:
As long as we don't mind the consistent use of repetitive card, note, issue, evidence case statements _everywhere_. I'll make that change too.

Reply:
1. Habit.
2. Send lets us use the power of ruby! Is really all. Otherwise we end up with:

```ruby
    case params['resource_type']
    when 'card'
      authorized_list_ids = List.where(board_id: current_project.boards.select(:id)).select(:id)
      @resource = Card.find_by(id: params['resource_id'], list_id: authorized_list_ids)
    when 'evidence'
      @resource = current_project.evidence.find_by id: params['resource_id']
    when 'issue'
      @resource = current_project.issues.find_by id: params['resource_id']
    when 'note'
      @resource = current_project.notes.find_by id: params['resource_id']
    end
```

Which I'm okay with. It's just repetitive.